fix(service-worker): detect new version even if files are identical to an old one

gkalpak opened the pull request:
## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] ~~Docs have been added / updated (for bug fixes / features)~~


## PR Type
```
[x] Bugfix
```

## What is the current behavior?
If an app version (aka manifest) contains the same files as an older version (e.g. making a change, then rolling it back), the SW would not detect it as the latest version (and update clients).

Issue Number: #24338


## What is the new behavior?
A new `timestamp` field is added in `ngsw.json` (based on the current date), which makes each build unique (with sufficiently high probability).


## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```


## Other information
Fixes #24338.

mary-poppins commented:
You can preview d12a118 at https://pr26006-d12a118.ngbuilds.io/.

mary-poppins commented:
You can preview 79fa1bc at https://pr26006-79fa1bc.ngbuilds.io/.

mary-poppins commented:
You can preview 94ad278 at https://pr26006-94ad278.ngbuilds.io/.

IgorMinar commented:
@gkalpak can you please rebase this to get rid of the pending pullapprove status? thanks

gkalpak committed:
fix(service-worker): detect new version even if files are identical to an old one

Previously, if an app version contained the same files as an older
version (e.g. making a change, then rolling it back), the SW would not
detect it as the latest version (and update clients).

This commit fixes it by adding a `timestamp` field in `ngsw.json`, which
makes each build unique (with sufficiently high probability).

Fixes #24338

gkalpak committed:
fixup! fix(service-worker): detect new version even if files are identical to an old one
